wait_for_deploy.py

```diff
@@ -45,9 +85,40 @@ async def wait_for_deploy(
             ["git", "rev-parse", f"origin/{watch_branch}"], cwd=working_dir
         )
         latest_hash = output.decode().strip()
-    while await fetch_release_hash(hash_url) != latest_hash:
+
+    if expected_version:
+        log.info(
+            "Expecting version '%s' for hash %s", expected_version, latest_hash[:7]
+        )
+    else:
+        log.info(
+            "No specific version expected for hash %s. Proceeding without version check.",
+            latest_hash[:7],
+        )
+
+    while True:
+        try:
+            current_hash = await fetch_release_hash(
+                hash_url, expected_version=expected_version
+            )
+            if current_hash == latest_hash:
+                log.info("Hash %s confirmed at %s", latest_hash[:7], hash_url)
+                break  # Hashes match, deploy successful
+
+            # If hashes don't match, log and continue loop
+            log.info(
+                "Waiting for hash %s at %s, currently %s",
+                latest_hash[:7],
+                hash_url,
+                current_hash[:7],
+            )
+        except Exception as e:  # pylint: disable=broad-except
+            log.error("Error checking deploy status at %s: %s", hash_url, e)
```

Comment:
Won't this get noisy with an error every 30s until the version matches? I wouldn't consider that an actual error (i.e. something that needs to be fixed).

Reply:
Good point, removing...